src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractPackage`2.cs

```diff
@@ -95,12 +96,12 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
                 Workspace.AdviseSolutionEvents(solution);
             }
 
-            LoadComponentsInUIContextOnceSolutionFullyLoaded(cancellationToken);
+            LoadComponentsInUIContextOnceSolutionFullyLoadedAsync(cancellationToken).Forget();
         }
 
-        protected override void LoadComponentsInUIContext(CancellationToken cancellationToken)
+        protected override async Task LoadComponentsAsync(CancellationToken cancellationToken)
         {
-            ForegroundObject.AssertIsForeground();
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
```

Comment:
I guess this will be changed to @sharwell new JTF abstraction we will have in Roslyn side once that is checked in?

Reply:
code above already use ComponentHost so, MEF should be available here.